istresearch/traptor
# Upgraded traptor.py to python 3

# For the Reviewer

When this is complete, you should approve the PR via github.

# For the Reviewee

<Reminder PR Title should be JIRA_NUMBER and Useful Description>

## Summary
Description of changes.

#### Release Note
<Concise sentence describing change>Required.

#### Breaking Changes
<Description for Techops of how to handle changes, migrations, updates>None.

## Quality Assurance

You have gathered the following items:

Given all of the items above, you have updated your Application ORT at the following locations:
- **Features and Alerting**: <link to app ORT>Required.
- **P4 Alerting**: <link to p4-alerting PR>Required.

## Testing and Verification

## Testing and Verification

*Steps to test your application for someone not familiar with it.* Required.

1. Configure Kafka without security to ensure this still works. Edit the kafka docker-compose service to make it available outside the docker network, as shown below.
```
  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_LISTENERS: PLAINTEXT://:9092
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://192.168.1.198:9092   <---- Your IP here.
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_CREATE_TOPICS: "traptor:1:1"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    depends_on:
      - zookeeper
```
2. Start Kafka and zookeeper 
```
$ docker-compose up -d zookeeper kafka
```
3. Use secure pipes to connect to redis.

4. Configure the Traptor settings for the Kafka writer in configs/settings.py.
```
KAFKA_HOSTS = '192.168.1.198'  #                                          <---- Use your machine IP here
KAFKA_BROKER_VERSION_FALLBACK = '0.9.0'
KAFKA_API_VERSION_REQUEST = True
...
KAFKA_SECURITY_PROTOCOL = None  #                               <---- Use None
...
```
5. Run the traptor.py file and look for confirmation a document was written.

6. Confirm data on the topic using a console consumer. Be sure to specify your machine IP in place of 192.168.1.198 in the command that follows.
```
$ docker run -it --network=host edenhill/kafkacat:1.6.0 -b 192.168.1.198:9092 -t traptor
```
